store/src/java-test/com/zimbra/cs/service/account/ChangePasswordTest.java

```diff
@@ -97,11 +108,30 @@ public void testBasicHandlerWrongAuth() throws Exception {
     }
 
     @Test
-    public void testNeedsAuth() throws Exception {
+    public void testNeedsAuth() {
         final ChangePassword handler = new ChangePassword();
         final Map<String,Object> context = Collections.emptyMap();
 
-        Assert.assertTrue("handler.needsAuth()", handler.needsAuth(context));
+        Assert.assertFalse("handler.needsAuth()", handler.needsAuth(context));
+    }
+    
+    @Test
+    public void testBasicHandlerWithResetPasswordAuthTokenUsageIfMustChangePasswordIsEnabled() throws Exception {
```

Comment:
Add a failure test, at least for the case where they don't pass any auth token?

Reply:
I didn't find any way to provide an auth token as a null.